part-1.py

```diff
@@ -4,17 +4,37 @@
 # the appropriate comment.
 
 # factorial
-
+def factorial(num):
+    if num < 0:
+        raise ValueError
+    elif num == 0:
+        return 1
+    else:
+        return num * factorial(num - 1)
 
 
 # reverse
-
+def reverse(text):
```

Comment:
👍 , just be aware that `text[:-1]` creates a copy of the original string, so this is O(n^2) for time/space complexity.